GitCommands/StringPool.cs

```diff
@@ -83,7 +83,8 @@ public string Intern(string source, int index, int length)
 
         private void Grow()
         {
-            _capacity *= 2;
+            // Double the capacity
+            _capacity <<= 1;
```

Comment:
There are only a handful of calls to `Grow` while the load process. There is no gain in perf, but it degrades readability.

Reply:
Not to mention possible negative value.
```c#
int value=int.MaxValue;
value<<=1;
```

Reply:
I didn't mean to include this commit. It's part of some other investigation into replacing mod with masking, which is an operation that happens for every add/lookup, unlike this resize line.

Yes it could overflow but so could multiplication. I don't think it's an issue in this case either way.

```
> i = int.MaxValue
2147483647
> i * 2
-2
```

Reply:
Huh.  I thought * would cause an overflow exception.  Thanks.  Learned something.

Reply:
Got it. Checked vs unchecked context.  
```c#
checked{
//cause overflow here will raise an exception.
}
```